doc/specs/#1620 - Query Refinement for Disambiguation.md

```diff
@@ -0,0 +1,220 @@
+Number Name       Id                   Source
+-------------------------------------------
+   1   ExampleA    Example.Example1     msstore
+   2   ExampleA    Example.Example2     winget
+   3   ExampleA    Example.Example22    msstore
+
+Please enter a row number, package name, ID, or query to refine search...
+> _
+```
+
+If the user enters a query, the flags should be honored.
+```pwsh
+# Refined the search to Name/ID with `ExampleA` in it.
+Number Name       Id                   Source
+-------------------------------------------
+   1   ExampleA    Example.Example1     msstore
+   2   ExampleA    Example.Example2     winget
+   3   ExampleA    Example.Example22    msstore
+
+Please enter a row number, package name, ID, or query to refine search...
+> --source winget
+Found ExampleA [Example.Example2] Version 1.0.0
+This application is licensed to you by its owner.
+...
+```
+
+If a query matches nothing then exit the CLI with the message that the query couldn't find any result.
+
+```pwsh
+Number Name       Id                   Source
+-------------------------------------------
+   1   Example    Example.Example1     msstore
+   2   Example2   Example.Example2     winget
+...
+   20  Example20  Example.Example20    msstore
+
+Please enter a row number, package name, ID, or query to refine search...
+> Example.Example1 --source winget
+No package found matching input criteria
+```
+
+## Capabilities
+
+### Telemetry
+Telemetry data should be sent around the following items
+* When a disambiguation list is shown
+* What query caused the disambiguation to be shown
+  
+The current telemetry will need to be adjusted to reflect the following case
+* If a user was able to disambiguate the package, the command result should be treated as a success
+
+### Accessibility
+
+The screen reader read the text properly as it reads the current implementation.
+
+### Security
+
+It should not have any security issues.
+
+### Reliability
+
+The user might enter the number incorrectly in rush and might need to kill the running installation process and restart it.
+
+### Compatibility
+
+This won't work in MinTTY. It's a [known issue](https://github.com/git-for-windows/build-extra/blob/main/ReleaseNotes.md?rgh-link-date=2022-10-21T03%3A13%3A26Z#known-issues) in Git-for-windows.
+
+### Performance, Power, and Efficiency
+
+## Potential Issues
+* There should be no effect of the `--silent` flag on the selector.
+* Having the `upgrade` command included in this would conflict with [#2627](https://github.com/microsoft/winget-cli/issues/2627)
+  
+  For example - `winget upgrade Microsoft` might have a match with hundreds of packages and use might not want to see this prompt. Although this is an opt-in feature, there might be a clash if both features are enabled.
+
+## Future considerations
+* Keyboard interactivity using <kbd>↑</kbd>, <kbd>↓</kbd>, and <kbd>Ener</kbd>
+
+* Currently, Winget does not support installing multiple packages in a single command.
```

Comment:
This has changed since the last iteration of the spec;

It should probably be updated so that each package which needs disambiguation goes through the disambiguation prompt at the start of the install flow.

Reply:
@Trenly are you talking about https://github.com/microsoft/winget-cli/pull/4282 this? If yes, then what needs to be considered?

Reply:
> @Trenly are you talking about #4282 this? If yes, then what needs to be considered?

No, I'm talking about #2861